bar.go

```diff
@@ -268,13 +268,15 @@ func (b *Bar) SetPriority(priority int) {
 // if bar is already in complete state. If drop is true bar will be
 // removed as well.
 func (b *Bar) Abort(drop bool) {
+	if drop {
+		b.container.dropBar(b) // It is safe to call this multiple times with the same bar
+	}
 	select {
 	case b.operateState <- func(s *bState) {
 		if s.completed == true {
 			return
 		}
 		if drop {
-			b.container.dropBar(b)
```

Comment:
I think we can wrap `true` path here into anonymous goroutine like:
```go
if drop {
  go func() {
    b.container.dropBar(b)
    b.cancel()
  }()
} else {
    ...
}
```

Reply:
I've tried this approach with your reproducer example and found no deadlocks.
Can you please check? If it works for you, can you please update this PR?

Reply:
Updated, tests are running in https://github.com/containers/podman/pull/11378 .

Reply:
Alternatively, AFAICS this is equivalent to simpler #102 (which I’m filing separately to run the tests separately as well).

Reply:
https://github.com/vbauerster/mpb/pull/102#discussion_r699548382